backend_new/src/services/application-csv-export.service.ts

```diff
@@ -0,0 +1,656 @@
+            csvHeaders
+              .map((header) => `"${header.label.replace(/"/g, `""`)}"`)
+              .join(',') + '\n',
+          );
+
+          // now loop over applications and write them to file
+          applications.forEach((app) => {
+            let row = '';
+            let preferences: ApplicationMultiselectQuestion[];
+            csvHeaders.forEach((header, index) => {
+              let multiselectQuestionValue = false;
+              let parsePreference = false;
+              let value = header.path.split('.').reduce((acc, curr) => {
+                // return preference/program as value for the format function to accept
+                if (multiselectQuestionValue) {
+                  return acc;
+                }
+
+                if (parsePreference) {
+                  // curr should equal the preference id we're pulling from
+                  if (!preferences) {
+                    preferences =
+                      app.preferences as unknown as ApplicationMultiselectQuestion[];
+                  }
+                  parsePreference = false;
+                  // there aren't typically many preferences, but if there, then a object map should be created and used
+                  const preference = preferences.find(
+                    (preference) => preference.multiselectQuestionId === curr,
+                  );
+                  multiselectQuestionValue = true;
+                  return preference;
+                }
+
+                // sets parsePreference to true, for the next iteration
+                if (curr === 'preferences') {
+                  parsePreference = true;
+                }
+
+                if (acc === null || acc === undefined) {
+                  return '';
+                }
+
+                // handles working with arrays, e.g. householdMember.0.firstName
+                if (!isNaN(Number(curr))) {
+                  const index = Number(curr);
+                  return acc[index];
+                }
+
+                return acc[curr];
+              }, app);
+              value = value === undefined ? '' : value === null ? '' : value;
+              if (header.format) {
+                value = header.format(value);
+              }
+
+              row += value ? `"${value.toString().replace(/"/g, `""`)}"` : '';
+              if (index < csvHeaders.length - 1) {
+                row += ',';
+              }
+            });
+
+            try {
+              writableStream.write(row + '\n');
+            } catch (e) {
+              console.log('writeStream write error = ', e);
+              writableStream.once('drain', () => {
+                console.log('drain buffer');
+                writableStream.write(row + '\n');
+              });
+            }
+          });
+
+          writableStream.end();
+        });
+    });
+  }
+
+  async maxHouseholdMembers(): Promise<number> {
+    // TODO: is there a way to filter on listingId here?
```

Comment:
is this TODO still needed?

Reply:
We aren't able to filter by listingID, but we know the applicationIDs so we can calculate it from that. I'll make that change!